src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,169 @@
+    Fetch weather data from meteo API given a specified location,
+    time range in "YYYY-MM-DD" format and climate models.
+    """
+    params = {
+        "latitude": latitude,
+        "longitude": longitude,
+        "daily": VARIABLES,
+        "start_date": start_date,
+        "end_date": end_date,
+        "models": climate_models,
+    }
+
+    return fetch_url(API_URL, params)
+
+
+def compute_yearly_mean_and_std(data: Dict) -> pd.DataFrame:
+    daily_data = data.get("daily", {})
+
+    df = pd.DataFrame(daily_data)
+
+    # Convert 'time' to datetime
+    df["time"] = pd.to_datetime(df["time"])
+
+    # Extract year and create a new column for it
+    df["year"] = df["time"].dt.year
+
+    # Delete time column
+    df = df.drop("time", axis=1)
+
+    # Group by year and calculate mean and standard deviation
+    yearly_mean_and_std_df = df.groupby("year").agg(["mean", "std"])
+
+    return yearly_mean_and_std_df.reset_index()
+
+
+def plot_mean_and_std(
+    climate_models: List[str],
+    parameter: str,
+    city: str,
+    climate_data: Dict,
+    df: pd.DataFrame,
+):
+    plt.figure()
+    plt.style.use("ggplot")
+    plt.subplots(figsize=(10, 6))
+    for model in climate_models:
+        column = f"{parameter}_{model}"
+        plt.errorbar(
+            df["year"],
+            df[column]["mean"],
+            yerr=df[column]["std"],
+            linestyle="-",
+            elinewidth=0.5,
+            marker="o",
+            markersize=2,
+            label=f"{model}",
+            capsize=1,
+        )
+        plt.fill_between(
+            df["year"],
+            df[column]["mean"] - df[column]["std"],
+            df[column]["mean"] + df[column]["std"],
+            alpha=0.3,
+        )
+
+    parameter_unit = climate_data["daily_units"][f"{parameter}_{climate_models[0]}"]
+    parameter_name = parameter.replace("_", " ").capitalize()
+    plt.xlabel("Year")
+    plt.ylabel(f"{parameter_name} ({parameter_unit})")
+    plt.title(
+        "Evolution of the Mean and Standard Deviation of\n "
+        + f"{parameter_name} using different climate models for {city}"
+    )
+    plt.legend()
+    plt.tight_layout()
+    plt.show()
+
 
 def main():
-    raise NotImplementedError
+    start_date = "1950-01-01"
+    end_date = "2050-01-01"
+    climate_models = [
+        "CMCC_CM2_VHR4",
+        "FGOALS_f3_H",
+        "HiRAM_SIT_HR",
+        "MRI_AGCM3_2_S",
+        "EC_Earth3P_HR",
+        "MPI_ESM1_2_XR",
+        "NICAM16_8S",
+    ]
+    parameter_to_plot = "temperature_2m_mean"
+
+    city_to_data = {}
+    for city, coordinates in COORDINATES.items():
+        latitude, longitude = coordinates["latitude"], coordinates["longitude"]
+        data = fetch_data_meteo_api(
+            latitude, longitude, start_date, end_date, climate_models
+        )
+        if data is None:
+            continue
+        city_to_data[city] = data
+
+    city_to_processed_data = {}
+    for city, data in city_to_data.items():
```

Comment:
Why not merging all these for loops?. Not really relevant here, but you are iterating the cities three times. What if we have 1e6 cities?